test/common/formatter/substitution_formatter_test.cc

```diff
@@ -202,6 +202,58 @@ TEST(SubstitutionFormatterTest, plainNumberFormatter) {
               ProtoEq(ValueUtil::numberValue(400)));
 }
 
+class StalledTimeSource : public TimeSource {
```

Comment:
Can you use TestTimeSystem here and use the advance helpers? That would save us adding a new test helper

Reply:
done